src/components/Chart.svelte

```diff
@@ -380,16 +381,46 @@
     return { x: x + dx, y: y + dy - h, w: w, h: h };
   }
 
-  export function fitData(shouldAnimate = false): void {
-    if (datasets.length === 0) {
+  // Fit viewport to the currently available datasets.
+  // As the store can take a while to update, the arguments here include extra datasets
+  // to be included (in case a new one was just selected) or excluded (de-selected)
```

Comment:
I am not sure what you mean here...  Which store is updating?  If there is some delay or lag being inserted somewhere, should we be using a callback to trigger this after said update has completed?